app/models/export.rb

```diff
@@ -0,0 +1,59 @@
+require "spreadsheet"
+require "stringio"
+require "tempfile"
+
+class Export
+  attr_reader :error_message
+
+  def initialize
+    return unless block_given?
+
+    begin
+      yield(self)
+    ensure
+      close
+    end
+  end
+
+  def stream_response(response)
+    @response = response
+
+    create_spreadsheet
+    tempfile.open
+
+    while str = tempfile.read(1024) # rubocop:disable Lint/AssignmentInCondition
+      response.stream.write(str)
+    end
+  ensure
+    tempfile.close
+  end
+
+  def close
+    @response.stream.close if @response
+    @tempfile.close! if @tempfile
+  end
+
+  def filename
+    @filename ||= "master_inventory_#{Time.zone.now.strftime('%Y%m%d%H%M%S')}.xls"
+  end
+
+  def tempfile
+    @tempfile ||= Tempfile.new(filename)
+  end
+
+  def create_spreadsheet
+    return if @created
+    begin
```

Comment:
The `begin` / `end` is unnecessary. You can add the `rescue`, and `ensure` at the method level and it will have the same effect (with the small side effect that `@created = true` would get called again, but I tend to favor dropping 2 lines and de-indenting the block, which I feel makes the code a tad more readable).